### DISCUSSION TITLE
Is it possible to treat the warning 'Component X was not imported, exported, or provided by MDXProvider as global scope' as an error?

### DISCUSSION THREAD

**Discussion by krillboi** [facebook/docusaurus#7116] [Q&A] [answered]
If some of my developers writing documentation forget to add an import statement for a React component they're using in their .mdx files, the above warning is printed out while building. I'd like to treat this as an error since we basically deployed something that didn't look the way we expected because of a missing import statement.

**Comment by Josh-Cena** [comment, marked as answer]
Doing this is non-trivial. See https://github.com/facebook/docusaurus/issues/2448

TL;DR: the log is directly logged from MDX and it silently transforms that to a div without a config option. Unless you pipe the stdout stream and parse the log yourself, there's no way for Docusaurus to control this...

**Reply by krillboi** [reply]
Alright, thanks for clarifying.
